observer.go

```diff
@@ -0,0 +1,15 @@
+package opentracing
+
+// Observer can be registered with the Tracer to recieve notifications
```

Comment:
@hkshaw1990 I think the comments are a bit too specific. I would keep it simpler:
```
// Observer can be registered with the Tracer to receive notifications about new Spans.
// Tracers are not required to support the Observer API. The actual registration of the
// observer is implementation specific, e.g. it might look like this:
//
//     observer := myobserver.New()
//     tracer := client.NewTracer(..., client.WithObserver(observer))
```

@bhs I prefer not to introduce an official API for registering the observer, because if they are passed to the Tracer constructor the internal data structure can be immutable and not require a mutex for access.